src/parser/parser_env.ml

```diff
@@ -357,12 +344,20 @@ let add_used_private env name loc =
 let consume_comments_until env pos = env.consumed_comments_pos := pos
 
 (* lookahead: *)
-let lookahead ~i env =
-  assert (i < maximum_lookahead);
-  Lookahead.peek !(env.lookahead) i
+let[@inline] lookahead_0 env = Lookahead.peek_0 !(env.lookahead)
+
+let[@inline] lookahead_1 env = Lookahead.peek_1 !(env.lookahead)
+
+let[@inline] lookahead ~i env =
+  match i with
+  | 0 -> lookahead_0 env
+  | 1 -> lookahead_1 env
+  | _ -> assert false
```

Comment:
Curious if you looked at generated code before adding the inline directives, since it seems pretty likely that ocamlopt would inline these definitions anyway.

Reply:
I guess this does not have impact on non-flambda mode, will remove the annotations to avoid confusions